syft/generic/abstract/hookable.py

```diff
@@ -0,0 +1,32 @@
+from functools import wraps
+
+
+def chain_call(obj, method_name, *args, **kwargs):
```

Comment:
this is a nice convenience function - although I'm not sure that we ever truly go more than one child deep (or two children deep in special circumstances) within any call. I do assume that you have a use for it in mind though.

Reply:
Yeah, we don’t usually reach very far into the chain yet, but having stared at the code for a while, it’s starting to look like it would get simpler/cleaner if we did. There are also a few ad-hog implementations of this pattern floating around.